sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -441,6 +537,17 @@ class Dataset[T] private(
   /** Collects the elements to an Array. */
   def collect(): Array[T] = rdd.collect()
 
+  /**
+   * (Java-specific)
+   * Collects the elements to a Java list.
+   *
+   * Due to the incompatibility problem between Scala and Java, the return type of [[collect()]] at
```

Comment:
I see, then we should have collect as list too.
On Nov 4, 2015 5:19 AM, "Wenchen Fan" <notifications@github.com> wrote:

> In sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala
> <https://github.com/apache/spark/pull/9358#discussion_r43841577>:
>
> > @@ -441,6 +537,17 @@ class Dataset[T] private(
> >    /** Collects the elements to an Array. */
> >    def collect(): Array[T] = rdd.collect()
> >
> > +  /**
> > +   * (Java-specific)
> > +   * Collects the elements to a Java list.
> > +   *
> > +   * Due to the incompatibility problem between Scala and Java, the return type of [[collect()]] at
>
> We can construct right type of array while calling RDD.collect, the
> problem is the interface. At java side the return type of RDD.collect()
> is java.lang.Object and we need to do a type cast, which is not friendly
> to users.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/9358/files#r43841577>.
>